### PR TITLE
Add QGEMM benchmark

### PR DESCRIPTION
**Pull request by chenfucn**
**Description**: 
Add QGEMM benchmark to measure integer gemm performance

### CODE REVIEW

File: onnxruntime/test/mlas/bench/bench_util.h [hunk 1]
```diff
@@ -5,10 +5,29 @@
...
+  std::uniform_real_distribution<double> distribution(static_cast<double>(min_value), static_cast<double>(max_value));
+  
+  std::vector<ElementType> r(N);
+  for (size_t i = 0; i < N; i++) {
+    r[i] = static_cast<ElementType>(distribution(generator));
+  }
+  return r;
+}
 
 std::vector<float> RandomVectorUniform(std::vector<int64_t> shape, float min_value, float max_value);
```
Comment (yufenglee) [inline onnxruntime/test/mlas/bench/bench_util.h:32]: There's already similar implementation for those random generation: [https://github.com/microsoft/onnxruntime/blob/8d737f977056444a307f1b7f0bcd402fba62d790/onnxruntime/test/common/tensor_op_test_utils.h#L28](https://github.com/microsoft/onnxruntime/blob/8d737f977056444a307f1b7f0bcd402fba62d790/onnxruntime/test/common/tensor_op_test_utils.h#L28). It'd better to leverage it

Comment (chenfucn) [reply onnxruntime/test/mlas/bench/bench_util.h:32]: I've tried it, had too pull too much dependencies into the projects. feel like the change is too big

Comment (chenfucn) [reply onnxruntime/test/mlas/bench/bench_util.h:32]: maybe we can consider moving that template into another smaller header file,

**Review by yufenglee — APPROVED** [review]
:shipit:

**Pull request merged**